src/DictList.elm

```diff
@@ -937,6 +947,77 @@ fromDict dict =
     DictList dict (Dict.keys dict)
 
 
+{-| Takes a key-fn and a list.
+Creates a `Dict` which maps the key to a list of matching elements.
+    mary = {id=1, name="Mary"}
+    jack = {id=2, name="Jack"}
+    jill = {id=1, name="Jill"}
+    groupBy .id [mary, jack, jill] == DictList.fromList [(1, [mary, jill]), (2, [jack])]
+-}
+groupBy : (comparable1 -> a -> comparable2) -> DictList comparable1 a -> DictList comparable2 (List a)
+groupBy keyfn list =
```

Comment:
I totally agree with you. Its not only that this is more parallel to DictExtra, but more important, this makes the function more useful for many usecases.